gms/gossiper.hh

```diff
@@ -84,6 +85,45 @@ class i_endpoint_state_change_subscriber;
 
 class feature_service;
 
+class gossip_query_token_status_response {
+public:
+    // The keys can be:
+    // - nr_known_nodes : Number of nodes this node knows about
+    // - status_token_hash: The hash of the gossip status and tokens for nodes
+    // - normal_token_hash: The hash of the tokens for nodes in NORMAL status
+    // - bootstrap_token_hash: The hash of the tokens for nodes in BOOTSTRAPPING status
+    // - leaving_nodes_hash: The hash of the nodes that are leaving the cluster
+    // - pending_ranges_hash: The hash of pending ranges
+    // - same_on_all_shards: If all the shards have the same hashes
+    static constexpr const char* nr_known_nodes = "nr_known_nodes";
+    static constexpr const char* status_token_hash = "status_token_hash";
+    static constexpr const char* normal_token_hash = "normal_token_hash";
+    static constexpr const char* bootstrap_token_hash = "bootstrap_token_hash";
+    static constexpr const char* leaving_nodes_hash = "leaving_nodes_hash";
+    static constexpr const char* pending_ranges_hash = "pending_ranges_hash";
+    static constexpr const char* same_on_all_shards = "same_on_all_shards";
+
+    // The values for same_on_all_shard
+    static constexpr const char* same_on_all_shards_yes = "yes";
+    static constexpr const char* same_on_all_shards_no = "no";
+
+    std::map<sstring, sstring> status;
```

Comment:
This should either be `private`, or the `class` should be a `struct`.

But even better, it should be changed to a set of different fields in the class as described in the other comment.

Reply:
idl needs to access the status. It is not a trival class, so it is better to be class instead of a struct. What's wrong with class + public member of status.

Reply:
Well it doesn't properly encapsulate/hide stuff as a typical `class` does. I believe that `struct`s are usually used for transparent data structures like this which expose their internals.

Reply:
For IDL you can use `friend struct ser::serializer<gossip_query_token_status_response`, I believe that's how it's usually done.

Reply:
The status is designed to be public. Why do we even bother to make it private and play the friend game.